### PR TITLE
terminal-util: use colon as separator for specifying color

### PR DESCRIPTION
**Pull request by yuwata**
Fixes a regression caused by https://github.com/systemd/systemd/commit/891abc9cf1d3d7afd7b740b4077551dcfb5f8e16.
Fixes https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1074073.
Fixes https://github.com/systemd/systemd/issues/33449.

### CODE REVIEW

**Comment by yuwata** [comment]
Please backport the first two commits to v256-stable.

File: src/basic/terminal-util.c [hunk 1]
```diff
@@ -1315,69 +1316,68 @@ bool terminal_is_dumb(void) {
...
-        if (r >= 0)
-                return r > 0 ? COLOR_ON : COLOR_OFF;
-        return _COLOR_INVALID;
-}
+                return _COLOR_MODE_INVALID;
 
-ColorMode get_color_mode(void) {
+        ColorMode m = color_mode_from_string(e);
+        if (m < 0)
+                return log_debug_errno(m, "Failed to parse $SYSTEMD_COLORS, ignoring: %m");
```
Comment (YHNdnzj) [inline src/basic/terminal-util.c]: Log the value here?

**Review by YHNdnzj — APPROVED** [review]

**Comment by yuwata** [comment]
@YHNdnzj Thank you for the review. The comment above is addressed. Upgrading the green label.

**Comment by poettering** [comment]
lgtm

**Comment by yuwata** [comment]
Please re-review. The original fix was dropped, and now ':' is used for separator.

**Pull request merged**